proposals/0012-f3-deals-reliability-ux.md

```diff
@@ -0,0 +1,159 @@
+
+Filecoin aims to become the storage layer for the decentralised future.
+The web3/dweb communities are excited, engaged, and traction is on the uptick everyday.
+
+Mining, consensus, onboarding storage capacity, and proving sectors, are all fully functional processes in the Filecoin Network, and namely in the reference implementation: [Lotus](https://github.com/filecoin-project/lotus). However, the object of this proposal, **the end-to-end storage deal process in Lotus,** falls short in terms of reliability and usability.
+
+Through various interactions with users and developers, we have confirmed these observations:
+
+1. users of Lotus struggle to understand the various deal stages and concepts.
+2. querying the network for status of an inflight deal is non-trivial, and requires too much protocol-specific knowledge.
+3. deal-making is a relatively long process, composed of various moving parts. When it fails midway, it's not easy to dig up the root cause.
+4. generally speaking, it is hard to strike successful deals on Filecoin mainnet due to a number of technical reasons that compound.
+
+Consequently, first-time users and devs who venture to experiment with Filecoin suffer an frictionful experience. Enough to turn away.
+
+We posit that **deal-making is the bread and butter of Filecoin**; *it is the reason the platform exists in the first place.* Thus it should work like a Swiss clock, and only fail exceptionally and/or for good reasons.
+
+> We decided to name this Project F3 in a wink to shooting to the same level of usability of Amazon S3 (albeit with notable differences).
+
+#### Assumptions & hypotheses
+
+_What must be true for this project to matter?_
+
+1. Users struggle to grasp the concepts and components that comprise the deal-making process end to end.
+2. Users don't have intuition of on what happens under the hood (not the details, but the major phases).
+3. New users desist, or end up frustrated.
+4. Attempting to strike a storage deal takes more than 1 attempt, even if it appears that we're doing everything right.
+5. Developers leave with the impression that Filecoin's basic user-facing functionality doesn't work.
+
+#### User workflow example
+
+_How would a developer or user use this new capability?_
+
+**This focuses on the UX elements (functional requirements), not so much on the technical reliability aspects (technical requirements).**
+
+Arguably, the most common use is storing a file in the Filecoin network, and later retrieving it. Implementing a one-stop interactive CLI command that orchestrates the end-to-end flow that goes from having a file to having a deal, with all the intermediate steps, would be a desirable workflow.
+
+`lotus store <path>`
+
+1. add file to the repo.
+2. display miners (miner directory in the future).
+3. select several miners.
+4. get a quotes from those miners.
+5. select a miner.
+6. initiate a deal.
+7. etc.
```

Comment:
This sounds like a workflow.  A couple of questions:
1. What are the UX implications if the lotus process dies part way through the workflow state?
2. Will the the workflow state persisted anywhere?  Per comment below from dirkmc@ it sounds like the deal state is stored somewhere.   I'm asking about this from how to potentially support "dashboard that one can point to a node to display the status and details of all deals" in the future.
(feel free to disregard if these comments/questions aren't relevant)